src/rpc.rs

```diff
@@ -141,17 +141,8 @@ impl From<u32> for BlockNumber {
     }
 }
 
-/// System properties for a Substrate-based runtime
-#[derive(serde::Serialize, Deserialize, Debug, Clone, PartialEq, Eq, Default)]
-#[serde(rename_all = "camelCase")]
-pub struct SystemProperties {
-    /// The address format
-    pub ss58_format: u8,
-    /// The number of digits after the decimal point in the native token
-    pub token_decimals: u8,
-    /// The symbol of the native token
-    pub token_symbol: String,
-}
+/// Arbitrary properties defined in the chain spec as a JSON object.
+pub type SystemProperties = serde_json::Map<String, serde_json::Value>;
```

Comment:
could this even be `serde_json::Value` instead to be more permissive?

Reply:
I could, but the type is specified to be `Map` on the substrate side so I think it should mirror that. I think it would need properties, it doesn't really make sense to have discrete values for the properties.